### PR TITLE
Improve examples

### PR DESCRIPTION
**Pull request by conradoplg**
- Add a ciphersuite-specific example to each crate
- Add code to `gendoc` to also copy the example

Closes https://github.com/ZcashFoundation/frost/issues/131, but we might want to use async to make the example clearer and to also add a DKG example

### CODE REVIEW

File: frost-ristretto255/README.md [hunk 1]
```diff
@@ -3,30 +3,98 @@ of signers (FROST).
...
 
 ```rust
+use std::{collections::HashMap, convert::TryFrom};
 use rand::thread_rng;
-use frost_ristretto255::*;
+use frost_ristretto255 as frost;
+
+let mut rng = thread_rng();
+let numsigners = 5;
+let threshold = 3;
```
Comment (conradoplg) [inline frost-ristretto255/README.md:20]: I need to change this to min/max_signers and then also copy the example to #164 after that merges so I'll wait for that to happen and update this PR

Comment (conradoplg) [reply frost-ristretto255/README.md]: Done

**Review by dconnolly — APPROVED** [review]
lgtm!

Should we run `gendoc` as part of CI? That way changes to `frost-ristretto255` rustdoc get applied to all others automatically

**Comment by dconnolly** [comment]
> lgtm!
> 
> Should we run `gendoc` as part of CI? That way changes to `frost-ristretto255` rustdoc get applied to all others automatically

I wonder if we can make the job make suggested changes by the suggested change mechanic vs a new commit 🤔

**Comment by conradoplg** [comment]
> > lgtm!
> > Should we run `gendoc` as part of CI? That way changes to `frost-ristretto255` rustdoc get applied to all others automatically
> 
> I wonder if we can make the job make suggested changes by the suggested change mechanic vs a new commit thinking

Yeah I'm not very fond of CI making commits. I think the easiest thing to do is making gendoc behave like rustfmt and return nonzero if it does makes changes, so that CI will fail if we forget to run it

**Comment by dconnolly** [comment]
> > > lgtm!
> > > Should we run `gendoc` as part of CI? That way changes to `frost-ristretto255` rustdoc get applied to all others automatically
> > 
> > 
> > I wonder if we can make the job make suggested changes by the suggested change mechanic vs a new commit thinking
> 
> Yeah I'm not very fond of CI making commits. I think the easiest thing to do is making gendoc behave like rustfmt and return nonzero if it does makes changes, so that CI will fail if we forget to run it

I like this 👍

**Pull request merged**